packages/openapi-fetch/src/index.js

```diff
@@ -233,6 +233,60 @@ export default function createClient(clientOptions) {
   };
 }
 
+class UrlCallForwarder {
+  constructor(client, url) {
+    this.client = client;
+    this.url = url;
+  }
+
+  GET(init) {
+    return this.client.GET(this.url, init);
+  }
+  PUT(init) {
+    return this.client.PUT(this.url, init);
+  }
+  POST(init) {
+    return this.client.POST(this.url, init);
+  }
+  DELETE(init) {
+    return this.client.DELETE(this.url, init);
+  }
+  OPTIONS(init) {
+    return this.client.OPTIONS(this.url, init);
+  }
+  HEAD(init) {
+    return this.client.HEAD(this.url, init);
+  }
+  PATCH(init) {
+    return this.client.PATCH(this.url, init);
+  }
+  TRACE(init) {
+    return this.client.TRACE(this.url, init);
+  }
+}
+
+const clientProxyHandler = {
+  // Assume the property is an URL.
+  get: (coreClient, url) => new UrlCallForwarder(coreClient, url),
```

Comment:
We could memoize `UrlCallForwarder` here on either:
- An internal map
- The proxied object itself (that would probably imply creating another intermediary object in `wrapAsPathBasedClient` to not pollute the core client).

However, I'm not sure it is worth the effort:

The [lazy getter technique](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get#smart_self-overwriting_lazy_getters) is primarily for statically known getter names. We can adjust it to work with proxies, but we'll still be left with an indirection through the proxy:

```ts
const clientProxyHandler = {
  // Assume the property is an URL.
  get: (coreClient, url) => {
    if (!(url in coreClient)) {
      // BAD: pollutes client, but OK for the example here.
      coreClient[url] = new UrlCallForwarder(coreClient, url);
    }
    return coreClient[url];
  }
}
```

Based on [optimizing JS](https://romgrk.com/posts/optimizing-javascript#4-avoid-indirection), on modern Chrome, Proxies lead to 30x slowdown (with trivial no-op indirection). At this point, I doubt that the additional object allocation is noticeable.

It feels like we're a bit stuck in this sub-optimal situation. Something we can do for users that want high-performance and path based calls is recommend pre-calculating the `UrlCallForwarder`:

```ts

const myMethod = client["/my-path"];

function performanceSensitive() {
  myMethod.GET(...);
}
```

Reply:
Hum, I just had an idea: We could potentially get around the proxy limitation by chaining prototypes strangely:

If we *inherit* (as in: prototype chain) from the proxy with the object we use to memoize the properties, that might mitigate the performance impact.

I'll give this a try.

Reply:
Just saw this improvement. Yeah I think this solves my concerns. I hate making “vibe-based” comments, especially when we don’t have solid benchmarking in the project. But once we give people a good DX like this, it’s tough to take it away again, and it’s easy to be stuck with any performance loss permanently.

But your explorations here seem to have found a happy medium of adding great DX but without incurring a noticeable perf hit (IMO having a proxy for 100% of calls would be a noticeable perf hit, and we managed to avoid that).

Overall I’m pretty happy with this approach!